### PR TITLE
Allow to remove project ownership to project creator

### PR DESCRIPTION
**Pull request by harminius**
Resolves https://github.com/MerginMaps/server-private/issues/2256

Alllow to remove ownership of the project owner.

### CODE REVIEW

**Comment by varmar05** [comment]
I found these places with creator used
- for template user (we can keep as is)
- for filters to check whether user workspace has some projects (legacy) and for filter of projects with 'created' flag (legacy) -> here I suggest to add explicit check for permissions (read) and not rely on creator flag

**Comment by varmar05** [comment]
also @harminius could you double check creator_id remains intact when project is deleted?

**Comment by MarcelGeo** [comment]
I think It should be good idea to remove ```creator``` attribute from api responses in EE and remain it just for internal use.

**Comment by varmar05** [comment]
> I think It should be good idea to remove `creator` attribute from api responses in EE and remain it just for internal use.

also we should mark public API response as deprecated and start removing it from clients (if used)

**Comment by harminius** [comment]
> I found these places with creator used
> 
> * for template user (we can keep as is)
> * for filters to check whether user workspace has some projects (legacy) and for filter of projects with 'created' flag (legacy) -> here I suggest to add explicit check for permissions (read) and not rely on creator flag

@varmar05 I think the `created` filter is only an additional filter for listing projects to which you need to have at least `read` access, no?

image

File: server/mergin/tests/test_private_project_api.py [hunk 1]
```diff
@@ -382,14 +382,14 @@ def test_update_project_access(client, diff_project):
...
-    # access of project creator can not be removed
+    # access of project creator can be removed
     data["user_id"] = diff_project.creator_id
     resp = client.patch(
         f"/app/project/{diff_project.id}/access",
         headers=json_headers,
         data=json.dumps(data),
     )
-    assert resp.status_code == 400
+    assert resp.status_code == 200
```
[the remark this answers is not in the document]
Comment (harminius) [reply server/mergin/tests/test_private_project_api.py:393]: :heavy_check_mark:

**Review by varmar05 — APPROVED** [review]

**Pull request merged**